rcl_interfaces/msg/FloatingPointRange.msg

```diff
@@ -1,14 +1,13 @@
 # Represents bounds and a step value for a floating point typed parameter.
 
 # Lower bound for valid values, inclusive.
-double minimum_value
+double from_value
 
 # Upper bound for valid values, inclusive.
-double maximum_value
+double to_value
 
-# Size of valid steps between the lower and upper bound.
+# Size of valid steps between the from and to bound.
 # Step should be positive, and a step of zero implies continuous values.
-# Step size should be less than the distance between the minimum and maximum
-# values, and ideally would be an even multiple of that distance, but that
-# is not required.
+# Step size should be less than the distance between the values, and ideally
```

Comment:
I think we should relax this as @dirk-thomas suggested in order to allow negative step values (my proposal is just to take the `abs`/`fabs` of step always, so something like this (I extended the documentation further as well to cover more corner cases):

> Step is considered to be a magnitude, therefore negative values are treated the same as positive values, and a step value of zero implies a continuous range of values.
> Ideally, the step would be both less than or equal to the distance between the bounds, as well as an even multiple of the distance between the bounds, but neither are required.
> If the absolute value of the step is larger than or equal to the distance between the two bounds, then the bounds will be the only valid values, e.g. if the range is defined as `{from_value: 1.0, to_value: 2.0, step: 5.0}` then the valid values will be `1.0` and `2.0`.
> If the step is less than then distance between bounds, but not a multiple of the distance, then the "to" bound will always be a valid value, e.g. if the range is defined as `{from_value: 2.0, to_value: 5.0, step: 2.0}` then the valid values will be `2.0`, `4.0`, and `5.0`.

I think the nice thing about describing as above is that all combinations of values that you can put in the message are valid, even if they change behavior GUI or something like that in odd cases.

Reply:
+1 for explaining cases with examples. Though, I think based on our offline discussion yesterday, we don't want to include `to_value` if the distance is not a multiple of `step`.

Reply:
Yes, that definitely needs to be changed.

Reply:
I don't fully understand why we don't want to linclude 'to_value' if the distance is not a multiple of 'step'.

Reply:
The idea is that the value of `to_value` is not valid.
E.g. let `from_value=0`, `to_value=5`, and `step=2`
Then, valid values for the parameter are `[0, 2, 4]`.
`5` is not a valid value, even though it is the value of `to_value`.

Reply:
The other case to document is `{from_value: 1.0, to_value: 2.0, step: 5.0}`

I'm not sure if we discussed this, but I think applying the same rule we are left with a single valid parameter value: `1.0`.

Reply:
> I don't fully understand why we don't want to linclude to_value if the distance is not a multiple of step.

From yesterday's discussion, I understand that it makes more sense and it's more clear to constraint the values to {`to_value` +/- n * `step`). In general terms you should be able to adjust `to_value` to comply with that constraint.

At least to me, in cases like:
- `from_value` = 0
- `step` = 4
- `to_value` = 5
I really don't see a practical use of including `to_value`. And if the step size is larger than `|to_value - from_value|`, then only `to_value` should be allowed.

Reply:
@jubeira I think that is almost correct. Since values go "from" -> "to", I think you mean to use {`from_value` +- n * `step`} as the constraint.

And,

```diff
- And if the step size is larger than `|to_value - from_value|`, then only `to_value` should be allowed.
+ And if the step size is larger than `|to_value - from_value|`, then only `from_value` should be allowed.
```

Reply:
Thanks @jacobperron, yes, that's exactly what I meant!

Reply:
What I have in my mind is that we state in the documentation that there is a `from_value` a `to_value` and a `step`. Then we state `Step size should be less than the distance between the values, and a multiple of that distance`. If those conditions are not met I think that the constraint as a whole should be rejected because it doesn't satisfy the required conditions.

Reply:
> If those conditions are not met I think that the constraint as a whole should be rejected because it doesn't satisfy the required conditions.

We didn't agree to that, as far as I understood our last meeting.

Reply:
Also you still need `to_value` in the case that `step` is `0`.

Reply:
> The other case to document is {from_value: 1.0, to_value: 2.0, step: 5.0}

> I'm not sure if we discussed this, but I think applying the same rule we are left with a single valid parameter value: 1.0.

I think we didn't decide on this either, but I'm not sure I agree with that conclusion. This is a consequence of deciding that `to_value` is not always a valid value, which was not what I was thinking in my original proposal, but is what we decided during the meeting.